packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -760,13 +760,18 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
 }
 
 class _AppBarState extends State<AppBar> {
+  final Map<String?, double> _scrollPositions = <String?, double>{};
   ScrollNotificationObserverState? _scrollNotificationObserver;
   bool _scrolledUnder = false;
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     _scrollNotificationObserver?.removeListener(_handleScrollNotification);
+    if (Scaffold.maybeOf(context)?.isDrawerOpen ?? false) {
```

Comment:
> Calling Scaffold.maybeOf to detect drawer and calling findAncestorStateOfType in the scroll notification is too expensive. Not a good idea to look up tree dozens of times as the user scrolls, if i remember correctly.

Thank you for remembering this @TahaTesser! 

@Mairramer I don't think we can do this then. since we tried it and had to revert it in the past.

Reply:
In fact, I believe we can. Because the solution implemented in the PR https://github.com/flutter/flutter/pull/122600/files put directly in the scroll notification. The path I chose was to place it inside didChangeDependencies.

Reply:
I made a change to avoid use findAncestorStateOfType too.

Reply:
@Mairramer This looks a lot better now. Good job.

Reply:
Scaffold.maybeOf uses findAncestorStateOfType

Reply:
Oh yes, but I believe that within didChangeDependencies we will avoid continuous checking, unlike a listener which would really be very expensive.

Reply:
Also, this does not seem to account for the end drawer? Although, it also looks like this ends up triggering scrolledUnder when it isn't. There is a google testing failure that shows an unexpected change, where where the app bar should not be the scrolled under color, but now it is.

Reply:
Oh, I forgot about him.